modules/core/include/opencv2/core/cvdef.h

```diff
@@ -384,8 +384,10 @@ typedef union Cv16suf
 {
     short i;
     ushort u;
-#if CV_FP16_TYPE
+#if CV_FP16_TYPE && defined __ARM_FP16_FORMAT_IEEE
     __fp16 h;
+#elif CV_FP16_TYPE // Other platforms suggested to use _Float16
+    _Float16 h;
```

Comment:
If I understand correctly, currently there is no x86_64 platform satisfying this case because `CV_FP16_TYPE` is enabled only for `__ARM_FP16_FORMAT_IEEE`. And in order to test this branch we need to also add support for `-mavx512fp16` and maybe some other option for RISC-V. Am I correct?